### ISSUE TITLE
[Feature Request] Allow downloading epub from archiveofourown.org

### ISSUE THREAD

**Issue by shinytea** [JimmXinu/FanFicFare#1110]
Archive of Our Own has a link that allows downloading of a story in epub format - as well as a few others.  I will admit, although the epub files that this program generates can be nice, I generally tend to prefer epub's that the AO3 itself generates, and would like to use this software merely to make sure my collection is up to date.

For example, this fic: https://archiveofourown.org/works/634290/chapters/1150509

Would have a download link shaped like this: https://archiveofourown.org/downloads/634290/A_Shot_in_the_Dark.epub?updated_at=1722040345

**Comment by JimmXinu** [comment]
IMO, FFF's epub output is better than AO3's.  

It does looks like AO3's epub is somewhat improved since that last time I looked--it appears to include images now.  It still generates the epub using Calibre Convert and discards the "work skin" CSS if applied by the author.

So I'd need a better reason than 'one user prefers AO3 epubs' to add an option to download a 'not as good' version.

But I'll leave this open for the moment in case there are other users or better reasons.

**Comment by shinytea** [comment]
One annoying thing I encountered in a story tonight was that a chatlog conversation between two people using e-mail address-like handles was messed up during the scraping process.

Apparently, AO3 uses Cloudflare protection to prevent e-mail scraping by browsers that lack Javascript.  An unintentional side effect is that anything in the story that looks like an e-mail address is turned into `[email_protected]` which links to [this page[(https://archiveofourown.org/cdn-cgi/l/email-protection).

**Comment by JimmXinu** [comment]
True.  Cloudflare has become very common now, unfortunately for us.

Can you share a story URL showing that? I don't have one handy.

**Comment by shinytea** [comment]
Sorry for the slow reply. The story I had in mind changed the `@` symbol to a lookalike symbol that didn't trigger AO3's protection.  However, I did find another "story" that seems to be an example of how to create a lookalike for an e-mail window, and this seems to trigger the protection when JS is turned off.

https://archiveofourown.org/works/7953412

**Comment by JimmXinu** [comment]
That was my thought, actually--to maybe use the HTML download if it wasn't obscured.  

I'm still not interested in adding download of AO3's epub to FFF--I just don't see enough benefit to shoe-horn it into FFF's paradigm.  At most, I see it saving the rare user who wants it a few clicks.

But I might be able to do something about `[email_protected]`...

**Comment by JimmXinu** [comment]
832387dea03db24ff6c6a0c7e0e4033d8774a2a9  Add `decode_emails:true` in `personal.ini` under `[defaults]` or a specific site section.  I've only tested with SpaceBattles and AO3--that's all the examples I could find in my library.  Only affects new downloads/updates.
